### PR TITLE
[Fixes AB#469] Append /discover on socials share

### PR DESCRIPTION
**Pull request by Skafec**

### CODE REVIEW

File: src/app/offers/offer-details/offer-details.component.ts [hunk 1]
```diff
@@ -25,7 +25,9 @@ export class OfferDetailsComponent implements OnInit {
     isOverview = false;
     isPortfolio = false;
     userConfirmed = true;
-    currentLocation = encodeURIComponent(window.location.href);
+    projectID = this.route.snapshot.params.id;
+    projectOverviewLink = window.location.host + '/overview/' + this.projectID + '/discover';
```
Comment (mj-ampnet) [inline src/app/offers/offer-details/offer-details.component.ts:29]: Prefer string interpolation over string addition:

```
projectOverviewLink = `${window.location.host}/overview/${this.projectID}/discover`
```

File: src/app/offers/offer-details/offer-details.component.ts [hunk 2]
```diff
@@ -25,7 +25,9 @@ export class OfferDetailsComponent implements OnInit {
     isOverview = false;
     isPortfolio = false;
     userConfirmed = true;
-    currentLocation = encodeURIComponent(window.location.href);
+    projectID = this.route.snapshot.params.id;
```
Comment (mj-ampnet) [inline src/app/offers/offer-details/offer-details.component.ts:28]: This cannot be inferred when the component is created! 

Snapshot should be _at the earliest_ inferred in `ngOnInit()` - take a look at the component lifecycle.

In general **don't work in declarations** - work in `constructor` or `ngOnInit` or other methods. Declarations should never be used for time sensitive stuff.

The order of operations might be completely wrong and how would you know?

Comment (mj-ampnet) [reply src/app/offers/offer-details/offer-details.component.ts:28]: Study this https://angular.io/guide/lifecycle-hooks#lifecycle-event-sequence

File: src/app/offers/offer-details/offer-details.component.ts [hunk 3]
```diff
@@ -25,7 +25,9 @@ export class OfferDetailsComponent implements OnInit {
...
+    projectOverviewLink = window.location.host + '/overview/' + this.projectID + '/discover';
+    currentLocation = encodeURIComponent(this.projectOverviewLink);
```
Comment (mj-ampnet) [inline src/app/offers/offer-details/offer-details.component.ts:30]: Don't refer to other properties in the declaration - this should be done in `constructor`

**Review by mightymatth — APPROVED** [review]

**Pull request merged**